lib/tree.rb

```diff
@@ -16,51 +16,137 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
-  def add(key, value)
-    raise NotImplementedError
+  # Time Complexity: O(logn)
+  # Space Complexity: O(logn)
+  def add(key, value = nil)
```

Comment:
👍 The time/complexities are correct if the tree is balanced and O(n) if unbalanced.